src/main/java/org/apache/mesos/hdfs/state/PersistentState.java

```diff
@@ -201,11 +217,14 @@ public void addHdfsNode(Protos.TaskID taskId, String hostname, String taskType)
   // Possibly call removeTask(slaveId, taskId) to avoid iterating through all maps
   public void removeTaskId(String taskId) {
     HashMap<String, String> journalNodes = getJournalNodes();
+    HashMap<String, String> journalNodeTaskNames = getJournalNodeTaskNames();
```

Comment:
At least as long as this pulls directly from zk, let's not call this until we're inside the if {} block where we know we'll actually use it, right before the remove call.